conda/cli/main.py

```diff
@@ -49,9 +52,37 @@ def main_subshell(*args, post_parse_hook=None, **kwargs):
     context.__init__(argparse_args=pre_args)
 
     parser = generate_parser(add_help=True)
+    
+    # HACK: this piece of code determines if conda should be trying to read an input --file
+    # arg as part of setting up the context. Conda should just do this for the following 
+    # commands.Other commands may also support a --file argument, but these files are not 
+    # environment spec files.
+    read_input_file = True if args[0] in ("env", "install", "create", "update") else False
```

Comment:
Currently the environment.yaml plugin will read any file that ends in `.yaml`/`.yml`. condarc config files are ones that have the name `condarc`, or have the file extension `.yml`/`.yaml`. This causes some awkward behaviour where conda will try to use the environment.yaml plugin to read a condarc file. This obviously does not work.

There are 2 options to resolve this:
* restrict the set of commands where conda tries to load a --file arg into the context (this solution); or
* make the [environment.yaml plugin more conservative](https://github.com/conda/conda-environments-project-mgmt/issues/68)

Reply:
Another question, what's `args` if we invoke `conda` like `conda --verbose create -d python`? This might not be robust enough. Do we have access to the subcommand invoked instead of relying on its position?

Reply:
Yep, this solution is absolutly not robust enough to handle any of the pre argument that conda has. So, specifying stuff like `conda --no-plugins env create --file env.yml` will not work as expected. I think the option of making the environment.yaml plugin more conservative is the best way to go. Will continue to explore other options too.